goose3/extractors/opengraph.py

```diff
@@ -41,7 +41,7 @@ def extract(self):
             if self.parser.getAttribute(meta, 'property') == "og:type"
         ]
         if og_types:
```

Comment:
It would be easier and more readable if the original if statement was changed:

``` python
if og_types and og_types[0]:
    og_type = og_types[0] + ':'
```

Reply:
Also, if we could get a test to check and ensure that it doesn't come back, that would be great. I say this because we should really be checking all the different values in the array and that is something I have been planning on adding for awhile.

Reply:
Fair enough. What if we did a simple membership test? Like so:
```
if og_types and None not in og_types:
    og_type = og_types[0] + ':'
```

Of course, this means that the entire block will be skipped if even one of the values in the list is None.

Reply:
Per the [OGP](http://ogp.me/#types) if `content` is not present, it really shouldn't be considered. The logic @barrust has would work and would allow the following to still fetch the `og:type`.

```html
<meta property="og:type" />
<meta property="og:type" content="website" />
```

The final `og_type` would be `website`. In addition, we could change the list comprehension above to not consider `NoneType`.

```python
og_types = [
    self.parser.getAttribute(meta, 'content')
    for meta in metas
    if self.parser.getAttribute(meta, 'property') == "og:type" and self.parser.getAttribute(meta, 'content')
]
```

Really any of the proposed changes to the logic should work though.

Reply:
@dlrobertson this is probably the best way of solving the problem. This also incorporates @barrust's suggestion of checking all the different values in the array.